pkg/helpers/helpers.go

```diff
@@ -321,3 +321,61 @@ func ManagedClusterAssetFn(fs embed.FS, managedClusterName string) resourceapply
 		return assets.MustCreateAssetFromTemplate(name, template, config).Data, nil
 	}
 }
+
+func TaintEqual(taint1 clusterv1.Taint, taint2 clusterv1.Taint) bool {
+	// Ignore the comparison of time
+	return taint1.Key == taint2.Key && taint1.Value == taint2.Value && taint1.Effect == taint2.Effect
+}
+
+func IsTaintsEqual(taints1 []clusterv1.Taint, taints2 []clusterv1.Taint) bool {
```

Comment:
A better implementation of this function is to build an index for the first taint array. The key is that taint itself without the `timeAdded`. And then try to match each element in the second taint array with the index.

Reply:
is this only used in testing? I feel you can use deep equal here.

Reply:
In the latest code, i didn' t remove this func because it is used in controller.

Reply:
nit: remove this add use deepequal in your test.